
Any tool relying on /dev/root has to be fixed.
Backport of dracut git commit 6d82a0470e7e340eb0353ecc3c524ded2e286e18

--- dracut-013/modules.d/99base/init.orig	2011-09-20 09:43:02.000000000 +0200
+++ dracut-013/modules.d/99base/init	2011-09-20 09:44:18.000000000 +0200
@@ -447,6 +447,9 @@
 
 wait_for_loginit
 
+# remove helper symlink
+[ -h /dev/root ] && rm -f /dev/root
+
 info "Switching root"
 
 unset PS4
